packages/client/src/features/tools/di.config.ts

```diff
@@ -23,15 +23,19 @@ import { DelKeyDeleteTool, MouseDeleteTool } from "./delete-tool";
 import { EdgeCreationTool } from "./edge-creation-tool";
 import { EdgeEditTool } from "./edge-edit-tool";
 import { NodeCreationTool } from "./node-creation-tool";
+import { MarqueeMouseTool } from "./marquee-mouse-tool";
+import { MarqueeTool } from "./marquee-tool";
 
 const toolsModule = new ContainerModule((bind, _unbind, isBound) => {
     // Register default tools
     bind(GLSP_TYPES.IDefaultTool).to(ChangeBoundsTool);
     bind(GLSP_TYPES.IDefaultTool).to(EdgeEditTool);
     bind(GLSP_TYPES.IDefaultTool).to(DelKeyDeleteTool);
+    bind(GLSP_TYPES.IDefaultTool).to(MarqueeTool);
```

Comment:
It probably makes sense to introduce a dedicated 'marqueeToolModule'
This module than binds the corresponding tools, and configures the modelelement/view.

Reply:
So that the marquee tool is no longer handled by 'tools' feature and is completly handled in a newly created 'marquee-tool' feature?
The marqueeToolModule would then look like this:
`const glspMarqueeToolModule = new ContainerModule((bind, unbind, isBound, rebind) => {
    const context = { bind, unbind, isBound, rebind };
    configureModelElement(context, 'marquee', MarqueeNode, MarqueeView);
    bind(GLSP_TYPES.IDefaultTool).to(MarqueeTool);
    bind(GLSP_TYPES.ITool).to(MarqueeMouseTool);
});`

Reply:
Correct, however on second thought it may be better to just create a utility function `configureMarqueeTool` which is then called inside of the tools module
`const toolsModule = new ContainerModule((bind, _unbind, isBound) => {
...
 const context = { bind, isBound};
configureMarqueeTool(context);
}

export function configureMarqueeTool((context: { bind: interfaces.Bind, isBound: interfaces.IsBound }){

    configureModelElement(context, 'marquee', MarqueeNode, MarqueeView);
    bind(GLSP_TYPES.IDefaultTool).to(MarqueeTool);
    bind(GLSP_TYPES.ITool).to(MarqueeMouseTool);
'

Reply:
Would you then expect the `MarqueeNode`, `MarqueeView`, `MarqueeMouseTool` and `MarqueeTool` to be in the tools feature or in the separate marquee-tool, even though they will be bound in the tools feature?

Reply:
I would omit the separate marque-tool for now and keep  MarqueeNode, MarqueeView, MarqueeMouseTool and MarqueeTool in the tools feature but extract the actual binding process into a utility function. This way it can be easily reused and/or removed in subclasses.